gulpfile.js

```diff
@@ -0,0 +1,872 @@
+var minifyShaders = taskName === 'minify' || taskName === 'minifyRelease' || taskName === 'release' || taskName === 'makeZipFile';
+
+var version = packageJson.version;
+if (/\.0$/.test(version)) {
+    version = version.substring(0, version.length - 2);
+}
+
+var sourceFiles = ['Source/**/*.js',
+                   '!Source/*.js',
+                   '!Source/Workers/**',
+                   '!Source/ThirdParty/Workers/**',
+                   'Source/Workers/createTaskProcessorWorker.js'];
+
+var buildFiles = ['Specs/**/*.js',
+                  '!Specs/SpecList.js',
+                  'Source/Shaders/**/*.glsl'];
+
+var jsHintFiles = ['Source/**/*.js',
+                   '!Source/Shaders/**',
+                   '!Source/ThirdParty/**',
+                   '!Source/Workers/cesiumWorkerBootstrapper.js',
+                   'Apps/**/*.js',
+                   'Apps/Sandcastle/gallery/*.html',
+                   '!Apps/Sandcastle/ThirdParty/**',
+                   'Specs/**/*.js',
+                   'Tools/buildTasks/**/*.js',
+                   'gulpfile.js'];
+
+var filesToClean = ['Source/Cesium.js',
+                    'Build',
+                    'Instrumented',
+                    'Source/Shaders/**.js',
+                    'Specs/SpecList.js',
+                    'Apps/Sandcastle/.jshintrc',
+                    'Apps/Sandcastle/jsHintOptions.js',
+                    'Apps/Sandcastle/gallery/gallery-index.js',
+                    'Cesium-*.zip'];
+
+var filesToSortRequires = ['Source/**/*.js',
+                           '!Source/Shaders/**',
+                           '!Source/ThirdParty/**',
+                           '!Source/Workers/cesiumWorkerBootstrapper.js',
+                           '!Source/copyrightHeader.js',
+                           '!Source/Workers/transferTypedArrayTest.js',
+                           'Apps/**/*.js',
+                           '!Apps/Sandcastle/ThirdParty/**',
+                           '!Apps/Sandcastle/jsHintOptions.js',
+                           'Specs/**/*.js',
+                           '!Specs/spec-main.js',
+                           '!Specs/SpecRunner.js',
+                           '!Specs/SpecList.js',
+                           '!Apps/Sandcastle/Sandcastle-client.js',
+                           '!Apps/Sandcastle/Sandcastle-header.js',
+                           '!Apps/Sandcastle/Sandcastle-warn.js',
+                           '!Apps/Sandcastle/gallery/gallery-index.js'];
+
+gulp.task('default', ['combine']);
+
+gulp.task('build', function(done) {
+    mkdirp.sync('Build');
+    glslToJavaScript(minifyShaders, 'Build/minifyShaders.state');
+    createCesiumJs();
+    createSpecList();
+    createGalleryList();
+    createSandcastleJsHintOptions();
+    done();
+});
+
+gulp.task('build-watch', function() {
+    gulp.watch(buildFiles, ['build']);
+});
+
+gulp.task('buildApps', ['combine', 'minifyRelease'], function() {
```

Comment:
I fixed this with a much simpler change (see my last submit).  I was missing a check for the `buildApps` task to determine if shaders should be minified or not. Of course I just realized that this causes shaders to be minified in both cases because the `build` step only runs once.  Was this the case before as well?  Or did we minify twice? (once with and without?)  A better solution is probably to do what you suggest, I'm just not sure how much bigger of a chance that will be.